packages/plugins/typed-variable-modal/src/TypedVariableModal.js

```diff
@@ -0,0 +1,368 @@
+/**
+ * @license
+ * Copyright 2020 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Class responsible for creating a modal used for creating typed
+ *     variables.
+ * @author aschmiedt@google.com (Abby Schmiedt)
+ */
+
+
+import * as Blockly from 'blockly/core';
+import { Modal } from './Modal.js';
+
+/**
+ * Class for displaying a modal used for creating typed variables.
+ */
+export class TypedVariableModal extends Modal {
+  /**
+   * Constructor for creating a typed variable modal.
+   * @param {!Blockly.WorkspaceSvg} workspace The workspace that the modal will
+   *     be registered on.
+   * @param {string} btnCallbackName The name used to register the button
+   *     callback.
+   * @param {Array<Array<string>>} types An array holding arrays with the
+   *     display name as the first value and the type as the second.
+   *     Ex: [['Penguin', 'PENGUIN'], ['Giraffe', 'GIRAFFE']].
+   * @param {TypedVarModalMessages} opt_messages The messages for a typed variable
+   *     modal.
+   */
+  constructor(workspace, btnCallbackName, types, opt_messages) {
+    const title = opt_messages ? opt_messages["TYPED_VAR_MODAL_TITLE"] :
```

Comment:
Why don't we use Blockly.Msg to house these instead?

Reply:
I think the idea was to make it standalone. We don't necessarily know what Blockly is going to do with its messages in the future and by making this separate it gives us the benefit of not having to update in the future. Also, because we are not planning on translating any of these messages I don't see the benefit we get from having it on Blockly.Msg.

Reply:
Fair enough.
If Blockly changes messages in the future, I would imagine we would continue to be backwards compatible with Blockly.Msg. 
The pro of using Blockly.Msg that I think you're missing with this is it makes it so the plugin is compatible with a developer's localization flow. 

Blockly uses Blockly.Msg to house localized messages, internally, but if a developer has additional blocks, fields, etc and wants to localize those, they would do so with their own localization flow (translatewiki, crowdin, etc), and the output of these processes is a JSON that is dumped into Blockly.Msg. In other words, if they add these additional messages to their flow, everything would automagically work if you use Blockly.Msg.

Reply:
Ok that makes sense. I changed so that the typed modal uses Blockly.Msg.